docs/building-a-form/README.md

```diff
@@ -12,14 +12,18 @@ Use this example `form.js` file to build a basic form.
 
 Your form is generated from a `form.js` file, along with a few other key configuration files.
 
-### [Adding widgets and definitions](adding-widgets-and-definitions.md)
-
-There are many common fields and widgets you can use to build forms. There are common types of definitions: `schema`/`uiSchema` objects and functions that return `schema`/`uiSchema` objects.
-
 ### [Available form features and usage guidelines](available-form-features-and-usage-guidelines.md)
 
 These form features are available in the US Forms System library. We've provided information about how to implement them in your form.
 
+### [Available widgets](available-widgets.md)
+
+These common widgets are included in the us-forms-system by default. Set these widgets in a config file while building your form.
+
+### [Common definitions](common-definitions.md)
+
+These definitions address form field patterns that are more complex than widgets and include common label text, validation, or field components.
```

Comment:
I think this description is slightly inaccurate; it's not that definitions are more complex than widgets, but rather they accomplish something completely different. Widgets are simply used to render the HTML form element for a given field. Widgets don't know anything about the specific field they are rendering. They don't know anything about the text being rendered, the specific data that field is collecting, the context in which that question is being asked, etc. Their only job is to figure out what kind of HTML form element to render, sometimes with some additional functionality to make sure the data that gets entered into that field is handled properly.

Definitions are chunks of the config that have been extracted into a separate file because that chunk of config might be used in more than one place and it's easier to import a common piece of code from a separate file than to write the same piece of config multiple times. For example, you might have multiple full name fields throughout the form. By extracting that chunk of the form config into a separate file, you can more easily use it in multiple places. But if you don't want to use definitions, you can also just write everything to the form config directly. Definitions know about the specific way that question is going to asked: what is the label text? What is the error message? When is that question hidden? Are there additional CSS classes we want to pass to that field to format it differently?

Hopefully that explanation is helpful! Given all of that, I would rewrite this little blurb to be:

"Definitions are chunks of the form config for specific types of questions that can be dropped into the main form config where needed. They are usually used only for `uiSchema`, and include things like label text, validation functions, error messages, and which widget to render."

Reply:
This reminds me that I also intended to incorporate the information from your comment in https://github.com/usds/us-forms-system/pull/182#discussion_r204173108:

> Widgets are React components that simply return specific HTML form elements. Definitions are chunks of schema (either regular schema or uiSchema) that can be plugged into the overall form config that describe aspects of the form.

I appreciate the careful thought about defining the difference between widgets and definitions. It's going to help a lot.